### ISSUE TITLE
Readable 'end' event never fires

### ISSUE THREAD

**Issue by macedigital** [BriteVerify/node-sftp-server#7]
Hi,

we're having a problem with this module when uploading files. The `Readable` stream passed into the callback for the 'writefile' session listener never emits 'end'. 

As far as I can tell, node v6.7.0 and v6.3.x are affected, see below code snippet:

````js
var SFTPServer = require('node-sftp-server');
var myserver = new SFTPServer();

myserver.listen(8022);

myserver.on('connect', function(context) {
  // ...
  context.accept(function(session) {
    // ... 
    session.on('writefile', function (path, readableStream) {
       readableStream.on('end', function() {
         console.log('expected result: it ends ...');
         // actual result: 'end' never fires
       });
    });
  });
});
````

Shouldn't it be possible to detect whether the `Readable` stream is done reading the upload? Could it be related to https://nodejs.org/dist/latest-v6.x/docs/api/stream.html#stream_additional_notes ?

**Comment by uberbrady** [comment]
Can you check out the PR #8, and see if that solves it for you? That may be the solution, or at least part of it.

**Comment by uberbrady** [comment]
The bulk of the testing I've ended up doing tends to be just piping the various streams around, and somehow `pipe()` manages to read and write files just fine to our streams. I suspect it's not dependent upon the `end()` event firing, and is looking at something else to determine if the stream is finished or not.

Having re-re-re-re-read the Streams documentation, it *seems* to imply that if `read()` returns `null`, that your stream is at EOF. You can try to give that a shot - look for a null coming back from `read()` - and do your end-event-management there.

Regardless, we *should* be emitting `end` when the stream ends; that sounds like a bug.

**Comment by uberbrady** [comment]
Just pushed up 0.1.2 to npm - let me know if that doesn't solve the problem.

**Comment by macedigital** [comment]
@uberbrady, many thanks, and sorry for the late reply. Yes, the issue is fixed in 0.1.2 :+1:
